### DISCUSSION TITLE
Connect rears of two fiber patch?

### DISCUSSION THREAD

**Discussion by aseques** [netbox-community/netbox#14017] [Q&A] [answered]
I have a run of 48 fibers between two locations on our company (LC on front / splice on back), I would like to connect them so I can trace the circuit from the switch port to the other extreme.
The logical way to connect them is to go to the device and connect both rear ports
![ksnip_tmp_rEsBtR](https://github.com/netbox-community/netbox/assets/555542/a463b481-0b9a-4f1b-bbfc-b2e91b20b113)
But I have to do it manually (it seems I can't multiselect the cables on the list)
![ksnip_tmp_LaxSRn](https://github.com/netbox-community/netbox/assets/555542/55ae134a-6cdb-4950-bb40-d34b125b0998)
Is that the proper way to do it? On the front port creation there was a similar issue where one could not multiselect the ports that was fixed in a later release, is it the same case?

**Comment by candlerb** [comment, marked as answer]
I wrote a Custom Script a while back which lets you connect cables in bulk:
https://github.com/netbox-community/customizations/blob/master/scripts/multi_connect.py

However it creates a single cable for each port connection; it doesn't create half the number of cables with 2 terminations on each. If you need to do it that way, you could use this code as a starting point.

Another option would be to prepare a list for CSV import, but importing a cable with multiple terminations also looks to be tricky / undocumented, and I couldn't work out how to do it.  You should be able to do it via the REST API though: see #12830 and #13280.

Finally, if all your patch panels are just linked back to back, an alternative approach is to use a multi-position rear port in your device type:
- create a single rear port with 48 "positions"
- create 48 front ports, where each front port N is linked to the single rear port with position N (this can be done in a single bulk operation, e.g. create `F[1-48]`)

Then you just need to create a single cable between the rear ports of two patch panels, job done.

Note that this isn't entirely accurate, even if you have a single MPO-MPO cable it may have crossed pairs (1-2 to 2-1, 3-4 to 4-3 etc).  Since Netbox doesn't distinguish between TX and RX legs on a cable, and doesn't even model separate TX/RX legs on an interface, this is unlikely to be a major problem.

**Comment by aseques** [comment]
hi @candlerb  thanks a lot for your answer, in fact my use case is the easiest one of the ones you wrote, because all the cables are 1 to 1
> Then you just need to create a single cable between the rear ports of two patch panels, job done.
Having to this one by one is a pain, so I ended up using your script that did the job perfectly.
One minor thing is that I was trying to run the script from my computer instead of netbox, so I lost quite a bit of time until I found [this guide](https://netboxlabs.com/zerotohero/11-custom-scripts/), so I will create a patch on the README.md to point to this page [here](https://docs.netbox.dev/en/stable/customization/custom-scripts/#running-custom-scripts)

Thanks a lot

**Reply by DanSheps** [reply]
In the future, please make sure you mark answers as answered.
